services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -1,29 +1,28 @@
 package cromwell.pipeline.service
 
-import java.util.UUID
-
 import cromwell.pipeline.datastorage.dao.repository.ProjectRepository
 import cromwell.pipeline.datastorage.dto._
 import cromwell.pipeline.model.wrapper.UserId
 import cromwell.pipeline.service.Exceptions.{ ProjectAccessDeniedException, ProjectNotFoundException }
 
+import java.util.UUID
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectService(projectRepository: ProjectRepository, projectVersioning: ProjectVersioning[VersioningException])(
   implicit executionContext: ExecutionContext
 ) {
 
   private[service] def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
-    projectRepository.getProjectById(projectId).flatMap(getForUserOrFail(_, userId))
+    projectRepository.getProjectById(projectId).flatMap(project => getForUserOrFail(project.toSeq, userId))
 
   def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] =
-    projectRepository.getProjectByName(namePattern).flatMap(getForUserOrFail(_, userId))
+    projectRepository.getProjectsByName(namePattern).flatMap(getForUserOrFail(_, userId))
 
-  private def getForUserOrFail(project: Option[Project], userId: UserId): Future[Project] =
-    project match {
-      case Some(project) if project.ownerId == userId => Future.successful(project)
-      case Some(_)                                    => Future.failed(new ProjectAccessDeniedException)
-      case None                                       => Future.failed(new ProjectNotFoundException)
+  private def getForUserOrFail(projects: Seq[Project], userId: UserId): Future[Project] =
+    projects.find(_.ownerId == userId) match {
+      case Some(project)             => Future.successful(project)
+      case None if projects.nonEmpty => Future.failed(new ProjectAccessDeniedException)
+      case None if projects.isEmpty  => Future.failed(new ProjectNotFoundException)
```

Comment:
Cool! Let's just empathize that the last case is the default one.
```suggestion
      case _  => Future.failed(new ProjectNotFoundException)
```
or
```suggestion
      case None  => Future.failed(new ProjectNotFoundException)
```